wpimath/src/main/native/thirdparty/sleipnir/src/optimization/solver/util/Filter.hpp

```diff
@@ -182,7 +182,16 @@ class Filter {
  private:
   Variable* m_f = nullptr;
   double m_μ = 0.0;
-  small_vector<FilterEntry> m_filter;
+  wpi::SmallVector<FilterEntry> m_filter;
+
+  template <typename T, unsigned N, typename Pred>
+  static typename wpi::SmallVector<T, N>::size_type erase_if(
+      wpi::SmallVector<T, N>& c, Pred pred) {
+    auto it = std::remove_if(c.begin(), c.end(), pred);
+    auto r = c.end() - it;
+    c.erase(it, c.end());
+    return r;
+  }
```

Comment:
This is a common enough operation that we should add this to the SmallVector header (probably in a separate PR).